source/payload-interfaces/index.rst

```diff
@@ -86,167 +109,207 @@ type EFI_RESOURCE_MEMORY_RESERVED.
-section 6.5 EFI_HOB_GUID_TYPE.
+Points to the SMBIOS table in SMBIOS 2.x format if GUID is ``gPldSmbiosTableGuid``.
 
-``AcpiTableAddress``
+DEVICE TREE
+^^^^^^^^^^^
 
-Point to the SMBIOS table entry point.
+The bootloader might pass Device Tree to the payload. So that the payload could get the platform information from the table.
 
-DEVICE TREE HOB
-~~~~~~~~~~~~~~~
+**GUID**
 
-The bootloader might pass Device Tree through the GUID HOB to the
-payload. So that the payload could get the platform information from
-the table.
+::
 
-**HOB GUID**::
+  gPldDeviceTreeGuid = {0x6784b889, 0xb13c, 0x4c3b, {0xae, 0x4b, 0xf, 0xa, 0x2e, 0x32, 0xe, 0xa3}}
 
-  #define    DEVICE_TREE_GUID    {0x6784b889, 0xb13c, 0x4c3b, {0xae, 0x4b, 0xf, 0xa, 0x2e, 0x32, 0xe, 0xa3}}
+**Structure**
 
-**Hob Interface Structure**::
+::
 
   #pragma pack (1)
 
-  //
-  // Bootloader Device Tree hob
-  //
   typedef struct {
-    EFI_HOB_GUID_TYPE     Header;
-    UINT64                DeviceTreeAddress;
-  } DEVICE_TREE_HOB;
+    PLD_GENERIC_HEADER   PldHeader;
+    UINT64               DeviceTreeAddress;
+  } PLD_DEVICE_TREE;
 
   #pragma pack()
 
 **Member Description**
 
-``Header``
+``PldHeader``
 
-Header.Name set to DEVICE_TREE_GUID. See section 6.5 EFI_HOB_GUID_TYPE.
+PldHeader.Revision is 0.
 
 ``DeviceTreeAddress``
 
 Point to the Device Tree entry point.
 
-Graphics information HOB
-~~~~~~~~~~~~~~~~~~~~~~~~
-
-If bootloader initializes the graphics device, the bootloader might
-report graphics mode and framebuffer information through
-EFI_PEI_GRAPHICS_INFO_HOB, and graphics hardware information
-through EFI_PEI_GRAPHICS_DEVICE_INFO_HOB.
-
-EFI_PEI_GRAPHICS_INFO_HOB and EFI_PEI_GRAPHICS_DEVICE_INFO_HOB provide the basic information
-for the graphics display. These HOBs are described in the *PI Specification.*
-
-Please refer Appendix 6.6 EFI_PEI_GRAPHICS_INFO_HOB and 6.7 EFI_PEI_GRAPHICS_DEVICE_INFO_HOB for the details.
-
-Serial Information HOB
-~~~~~~~~~~~~~~~~~~~~~~
+Serial Information
+^^^^^^^^^^^^^^^^^^
 
 If the debug device type and subtype are specified in DBG2, the
-bootloader should pass SERIAL_PORT_INFO hob to payload. This hob
-provides 16550 compatible serial debug port information from
-bootloader to payload.
+bootloader should 16550 compatible serial debug port information
+to payload.
 
 **Opens: Should we let bootloader provide debug callback** **for debug?**
 
-**HOB GUID**::
+**GUID**
+
+::
 
-  #define    SERIAL_INFO_GUID    {0xaa7e190d, 0xbe21, 0x4409, {0x8e, 0x67, 0xa2, 0xcd, 0xf, 0x61, 0xe1, 0x70}}
+  gPldSerialPortInfoGuid   = {0xaa7e190d, 0xbe21, 0x4409, {0x8e, 0x67, 0xa2, 0xcd, 0xf, 0x61, 0xe1, 0x70}}
 
-**Hob Interface Structure**::
+**Structure**
+
+::
 
   #pragma pack(1)
 
   typedef struct {
-    UINT16     Reversion;
-    BOOLEAN    UseMmio;
-    UINT8      RegisterWidth;
-    UINT32     BaudRate;
-    UINT64     RegisterBase;
+    PLD_GENERIC_HEADER   PldHeader;
+    UINT16               Revision;

```

Comment:
Revision needs to be removed since it is covered by PLD_GENERIC_HEADER.

Reply:
yes.